src/sigma/sigma_primitives.hpp

```diff
@@ -71,34 +63,33 @@ std::vector<uint64_t> SigmaPrimitives<Exponent, GroupElement>::convert_to_nal(
 }
 
 template<class Exponent, class GroupElement>
-void SigmaPrimitives<Exponent, GroupElement>::get_x(
-        const GroupElement& A,
-        const GroupElement& C,
-        const GroupElement& D,
+void SigmaPrimitives<Exponent, GroupElement>::generate_challenge(
+        const std::vector<GroupElement>& group_elements,
         Exponent& result_out) {
     CSHA256 hash;
-    unsigned char data[3 * A.serialize_size];
-    unsigned char* current = A.serialize(data);
-    current = C.serialize(current);
-    D.serialize(current);
-    hash.Write(data, 3 * A.memoryRequired());
+    std::vector<unsigned char> data(group_elements.size() * group_elements[0].memoryRequired());
```

Comment:
This possible to crash if `group_elements` is empty.

Reply:
It can not be empty. Added a small check.